src/tests/Microsoft.PowerFx.Interpreter.Tests/TypeCoercionTest.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.PowerFx.Core.Tests;
+using Microsoft.PowerFx.Interpreter.Marshal;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Tests
+{
+    // Test type coercion from FormualValue to target type 
+    public class TypeCoercionTest : PowerFxTest
+    {
+        // From number to other types
+        [Theory]
+        [InlineData(1, "true", "1", "1", "12/31/1899 12:00 AM")]
+        [InlineData(0, "false", "0", "0", "12/30/1899 12:00 AM")]
+        [InlineData(44962, "true", "44962", "44962", "2/5/2023 12:00 AM")]
+        public void TryCoerceFromNumberTest(double value, string exprBool, string exprNumber, string exprStr, string exprDateTime)
+        {
+            TryCoerceToTargetTypes(FormulaValue.New(value), exprBool, exprNumber, exprStr, exprDateTime);
+        }
+
+        // From string to other types
+        [Theory]
```

Comment:
Does "True"  (capital t) coerce to true? (we should do whatever Boolean() does)

Reply:
will add that test.